src/append.jl

```diff
@@ -0,0 +1,168 @@
+record batches simultaneously (e.g. if julia is started with `julia -t 8`
+or the `JULIA_NUM_THREADS` environment variable is set).
+
+Supported keyword arguments to `Arrow.append` include:
+  * `compress`: possible values include `:lz4`, `:zstd`, or your own initialized `LZ4FrameCompressor` or `ZstdCompressor` objects; will cause all buffers in each record batch to use the respective compression encoding
+  * `alignment::Int=8`: specify the number of bytes to align buffers to when written in messages; strongly recommended to only use alignment values of 8 or 64 for modern memory cache line optimization
+  * `dictencode::Bool=false`: whether all columns should use dictionary encoding when being written; to dict encode specific columns, wrap the column/array in `Arrow.DictEncode(col)`
+  * `dictencodenested::Bool=false`: whether nested data type columns should also dict encode nested arrays/buffers; other language implementations [may not support this](https://arrow.apache.org/docs/status.html)
+  * `denseunions::Bool=true`: whether Julia `Vector{<:Union}` arrays should be written using the dense union layout; passing `false` will result in the sparse union layout
+  * `largelists::Bool=false`: causes list column types to be written with Int64 offset arrays; mainly for testing purposes; by default, Int64 offsets will be used only if needed
+  * `maxdepth::Int=$DEFAULT_MAX_DEPTH`: deepest allowed nested serialization level; this is provided by default to prevent accidental infinite recursion with mutually recursive data structures
+  * `ntasks::Int`: number of concurrent threaded tasks to allow while writing input partitions out as arrow record batches; default is no limit; to disable multithreaded writing, pass `ntasks=1`
+"""
+function append end
+
+append(file::String; kw...) = x -> append(file::String, x; kw...)
+
+function append(file::String, tbl;
+        largelists::Bool=false,
+        compress::Union{Nothing, Symbol, LZ4FrameCompressor, ZstdCompressor}=nothing,
+        denseunions::Bool=true,
+        dictencode::Bool=false,
+        dictencodenested::Bool=false,
+        alignment::Int=8,
+        maxdepth::Int=DEFAULT_MAX_DEPTH,
+        ntasks=Inf)
+    if ntasks < 1
+        throw(ArgumentError("ntasks keyword argument must be > 0; pass `ntasks=1` to disable multithreaded writing"))
+    end
+    if compress === :lz4
+        compress = LZ4_FRAME_COMPRESSOR
+    elseif compress === :zstd
+        compress = ZSTD_COMPRESSOR
+    elseif compress isa Symbol
+        throw(ArgumentError("unsupported compress keyword argument value: $compress. Valid values include `:lz4` or `:zstd`"))
+    end
+
+    open(file, "r+") do io
+        append(io, tbl, largelists, compress, denseunions, dictencode, dictencodenested, alignment, maxdepth, ntasks)
+    end
+
+    return file
+end
+
+function append(io, source, largelists, compress, denseunions, dictencode, dictencodenested, alignment, maxdepth, ntasks)
+    bytes = Mmap.mmap(io)
```

Comment:
Hmmm, I don't think this works if you provide a filename? Because you'd be calling `bytes = Mmap.mmap(io::IOStream)`? maybe that's supported? I'd rather follow the pattern we have in `Arrow.Table` where we call `Mmap.mmap` if they provide a `string`, otherwise we call `read(io)` to get the IPC stream bytes.

Reply:
addressed in https://github.com/JuliaData/Arrow.jl/pull/160/commits/00da06d181c9f2d385964cd450315dcbcb5f455b